### PR TITLE
Warn when ignore-tidy-linelength is present, but no lines are too long

### PR DESCRIPTION
**Pull request by varkor**
It's easy for a `// ignore-tidy-linelength` to be added when there is a genuine need to ignore a file's line length, but then after refactoring the need is gone, but the tidy directive is not removed. This means that in the future, further editing may accidentally add unnecessarily long lines. This change forces `// ignore-tidy-linelength` to be used exactly when necessary, to make sure such changes are intentional.

### CODE REVIEW

File: src/test/ui/borrowck/borrowck-unary-move.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
Comment (kennytm) [inline src/test/ui/borrowck/borrowck-unary-move.rs]: Is this change intended?

File: src/test/ui/cannot-mutate-captured-non-mut-var.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
Comment (kennytm) [inline src/test/ui/cannot-mutate-captured-non-mut-var.rs]: (Ditto)

File: src/test/ui/regions/region-borrow-params-issue-29793-big.rs [hunk 3]
```diff
@@ -6,6 +6,9 @@
 // behavior (because the improperly accepted closure was actually
 // able to be invoked).
 
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
Comment (kennytm) [inline src/test/ui/regions/region-borrow-params-issue-29793-big.rs]: (Ditto)

File: src/test/ui/borrowck/borrowck-unary-move.rs (hunk 1 above)
Comment (varkor) [reply src/test/ui/borrowck/borrowck-unary-move.rs]: Whoops, no, I must have messed up the rebase.

**Comment by kennytm** [comment]
r=me after Travis passes.

**Comment by petrochenkov** [comment]
This kind of logic is applicable to all tidy rules.

`// ignore-tidy-x` && `not x` -> tidy error

**Comment by varkor** [comment]
@kennytm: I've updated the PR to also warn when other tidy ignore directives are present, but have no effect. Could you check to make sure the changes are still okay?

**Comment by kennytm** [comment]
@bors r-

This PR failed in the rollup #60189, because #60172 has accidentally removed the CRLF line endings from [src/test/run-pass/issues/issue-16278.rs](https://github.com/rust-lang/rust/blob/0f11354a9c1bf0c5ac250c7fa2bafc289a662f42/src/test/run-pass/issues/issue-16278.rs) (see #16278). They need to be added back.

Screenshot 2019-04-23 at 17 50 26-fs8

**Comment by varkor** [comment]
> This PR failed in the rollup #60189, because #60172 has accidentally removed the CRLF line endings 

Ah, this was very easy to miss — the line endings were converted silently as soon as I modified the file 😬
I've reverted the change and am checking everything now passes again.
